thorlcr/thorutil/thormisc.cpp

```diff
@@ -1703,3 +1705,36 @@ void saveWuidToFile(const char *wuid)
     wuidFileIO->write(0, strlen(wuid), wuid);
     wuidFileIO->close();
 }
+
+bool hasTLK(IDistributedFile &file, CActivityBase *activity)
+{
+    unsigned np = file.numParts();
+    IDistributedFilePart &part = file.queryPart(np-1);
+    bool keyHasTlk = strisame("topLevelKey", part.queryAttributes().queryProp("@kind"));
+    if (!keyHasTlk)
+    {
+        // See HPCC-32845 - check if TLK flag is missing from TLK part
+        // It is very likely the last part should be a TLK. Even a local key (>1 parts) has a TLK by default (see buildLocalTlks)
+        if (np>1)
+        {
+            RemoteFilename rfn;
+            part.getFilename(rfn);
+            StringBuffer filename;
+            rfn.getPath(filename);
+            Owned<IKeyIndex> index = createKeyIndex(filename, 0, false, 0);
```

Comment:
I think in practice all indexes > 1 part will/should have TLK since they are by def. created by Thor and default with buildLocalTlks default on.

It could workout that the group is N and this is N+1 so means it really is an index with a TLK - and only check then, but don't think it's worth it?